docs/staking.md

```diff
@@ -0,0 +1,122 @@
+
+To accept staked funds, a delegate needs to have opted in to staking. This is done by `set_delegate_parameters`. This part is not supported by Taquito, as it is not a dApp operation. More information [here](https://tezos.gitlab.io/paris/adaptive_issuance.html#staking-policy-configuration).
+
+# Staking Funds
+To stake your funds, you need to call the `stake` operation.
+You need yo have delegated your funds to a baker before you can stake them.
+
+```javascript
+
+// TODO: Add code after the implementation is done
+
+```
+
+# Unstaking Funds
+
+To unstake your funds, you need to call the `unstake` operation. This will change your fund's status to "Unstaked +  Frozen"
+The unstaked funds will still be frozen (and subject to slashing for 4 cycles). After that, your funds will be in "Unstaked + Finalizable" state.
+
+```javascript
+
+```
+
+# Finalizing Unstake
+
+To finalize your unstaked funds, you need to call the `finalize_unstake` operation. This will change your fund's status to "spendable".
+
+```javascript
+
+```
+
+# Additional Info and Some Notes
+
+## Cycle
+At the time of this writing, a cycle is 16384 blocks, and 15 seconds per block. This means a cycle is about 2.8 days. This changes with the activation of newer protocols.
+
+## Overstaking
+A delegate can set the maximum amount of staking they can accept, as a multiply of their own balance. If a delegate's limit is exceeded, the exceeding stake is automatically considered as delegation for the delegate's baking and voting power calculation, but it does remain slashable. That means it's your responsibility to make sure you're not overstaking.
+
+```javascript
+// Todo: check for overstaking
+```
+
+## Changing Delegate
+When you change your delegate, your funds will be automatically unstaked. You will have to wait for 4 cycles before you can stake them again.
+
+## No need to finalize_unstake in some cases
+The `stake` and `unstake` operations will automatically finalize all currently finalizable funds.
+
+## Adaptive Issuance
+A concept related to staking is adaptive issuance. Adaptive issuance is a mechanism that adjusts the block reward based on the total amount of staked tez. In short, the reward will be adjusted to keep the staked funds about 50% of the total supply. To read more about the mechanism you can check [this document](https://tezos.gitlab.io/paris/adaptive_issuance.html#adaptive-issuance).
+
+## Staking Rewards
+In delegation, rewards were given to the baker, and it was the baker's responsibility to distribute the rewards to the delegators. In staking, the rewards are given directly to the staker. However, rewards are not given immediately. // TODO: how long does it take for rewards to be given?
+
+## Feature activation
+Voting for Staking is being done differently from other protocol updates. It is included in the "Oxford" protocol, and after "Paris" is activated, this feature can be voted on separately.
```

Comment:
paris have ai automatic activated, but it will be launch in cycle 6 of start of paris